### PR TITLE
Updated requirements.txt files

### PR DESCRIPTION
**Pull request by Jaybsoni**
Updated the `requirements.txt` file by pinning the version #s for the dependancies. This way we ensure that users can use `pip install -r requirements.txt` and get a stable version of the plugin. A similar change is made for the requirements file in the docs folder.

Finally, a new `requirements-ci.txt` is added for our ci pipeline so that we can install the latest version of the dependancies and update the plugins accordingly.

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,74 @@
-docutils==0.16
-sphinxcontrib-bibtex
-pygments-github-lexers
-cirq
-pennylane
-sphinx-automodapi
```
Comment (Jaybsoni) [inline doc/requirements.txt:6]: I can make the docs locally, but for some reason, build the docs fails. Also some of the dependencies listed here are not required for me to build the docs locally ...

File: doc/requirements.txt [hunk 2]
```diff
@@ -1,6 +1,74 @@
...
+jsonschema==3.2.0
+jupyter-core==4.8.1
+kiwisolver==1.3.2
+MarkupSafe==2.1.1
+matplotlib==3.4.3
+mpmath==1.2.1
+nbformat==5.1.3
+networkx==2.6.3
+ninja==1.10.2.3
+numpy==1.22.3
```
Comment (rmoyard) [inline doc/requirements.txt:33]:
```suggestion
numpy==1.21.6
```

**Review by rmoyard — COMMENTED** [review]
You should be able to pass the doc ci check, you just need to adapt some dependencies number e.g. numpy. If you are not sure go on action and take a look at the previous successful one what was the version number of failing packages.

**Comment by antalszava** [comment]
[sc-17333]

File: requirements-ci.txt [hunk 3]
```diff
@@ -0,0 +1,56 @@
+appdirs
```
Comment (rmoyard) [inline requirements-ci.txt:1]: Here as discussed in the meeting, you should have only direct dependencies ```
```
pennylane
cirq-core
cirq-pasqal
numpy
```

**Review by rmoyard — COMMENTED** [review]
Thanks @Jaybsoni, it looks good expect the `requirements-ci.txt, look at my related comment.

File: requirements-ci.txt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+scipy
+semantic-version
+six
+sortedcontainers
+sympy
+toml
+tqdm
+traitlets
+typing_extensions
+urllib3
```
Comment (Jaybsoni) [inline requirements-ci.txt:56]:
```suggestion
pennylane
cirq-core
cirq-pasqal
numpy
```

**Review by rmoyard — APPROVED** [review]
Thanks @Jaybsoni, it looks good to me 💯

**Pull request merged**